### PR TITLE
Extensible support for language runtimes

### PR DESCRIPTION
**Pull request by spmcbride1201**
This PR attempts to cleanup the architecture for providing support for alternate runtimes. It moves runtime-specific UI components into distinct directories.

To view JVM:
1. Open a terminal session to the project directory and run `npm run mock-jvm`
2. Open `/public/index.html` and ensure runtime is set to `JVM` on line 15
3. Open `package.json` and ensure that proxy is set to `http://localhost:9990`
4. Open a terminal session to the project directory and run `npm start`

You should see something like the following:
image

To view Go: 
1. End all terminal processes used to show JVM
2. Open a terminal session to the project directory and run `npm run mock-go`
3. Open `/public/index.html` and ensure runtime is set to `GOLANG` on line 15
4. Open `package.json` and ensure that proxy is set to `http://localhost:9092`
5. Open a terminal session to the project directory and run `npm start`

You should see something like the following. Notice that this dashboard has Percent CPU utilized, but the JVM dashboard did not. This is because this is a new metric that was not offered by Finagle.
image

## Known Regressions:
* The "drawers" of the Sidebar cards dynamically generated from JSON do not open as expected
* The React Router NavLInks do not update immediately, but hang until the next render loop from React. This suggests that something is wrong in a component lifecycle method.

### CODE REVIEW

File: config/webpack.config.dev.js [hunk 1]
```diff
@@ -188,7 +188,8 @@ module.exports = {
           {
             loader: require.resolve("css-loader"),
             options: {
-              importLoaders: 1
+              importLoaders: 1,
```
Comment (spmcbride1201) [inline config/webpack.config.dev.js]: All of the changes in this file are just to enable scss source maps in dev mode.

File: src/components/AppBrandBar.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const AppVersion = "0.3.3";
 const BrandName = "Fabric";
 const BrandLogo = decipherLogo;
 
+/** Sidebar Branding and Versioning */
```
Comment (spmcbride1201) [inline src/components/AppBrandBar.js]: It seems that I've been randomly adding new inline comments. To some degree, this was random, but I plan to go back and have a documentation PR. Since I'm refactoring so much right now and I've got different feature branches that work slightly differently, documentation seems pretty critical

File: src/components/Container.js [hunk 3]
```diff
@@ -38,36 +43,79 @@ class Container extends Component {
...
+    }
+  }
+
   render() {
     return (
       <div id="app-container">
-        <nav className="app-sidebar">
+        {/* Note: the following backgroundImage tag was inlined to get the URL to properly resolve.*/}
+        {/* This is currently resulting in a flicker on initial load, as the inline styles load first.*/}
+        <nav
```
Comment (spmcbride1201) [inline src/components/Container.js]: The naming of all of these sidebar components is pretty bad. I plan to go back to try to clean it up and refactor into a top level Sidebar component containing.

File: src/components/DataTotals.js [hunk 4]
```diff
@@ -7,6 +7,12 @@ DataTotals.propTypes = {
   sentBytes: PropTypes.number.isRequired
 };
 
+/**
+ * Simple card component that expresses total uploads and downloads
+ * @param {Object} props 
+ * @param {number} props.receivedBytes 
+ * @param {number} props.sentBytes
```
Comment (spmcbride1201) [inline src/components/DataTotals.js]: I believe that this is the properly way to annotate a complex object passed as an arg. It doesn't seem to feel into VSCode IntelliSense as I expected though, so it could be wrong though.

File: src/workers/ajax.js [hunk 5]
```diff
@@ -27,6 +27,26 @@ export function main(message) {
             .then(data => _.omitBy(data, value => Number.isNaN(value)))
         );
       }
+    // Temporaryily commented out pending further commit
```
Comment (spmcbride1201) [inline src/workers/ajax.js]: This is dead code from merging in stuff from `fabric-wide-dash` (a different feature branch that I had to put on hold). Previously, I allowed the scraping of multiple endpoints, but I tightened this down to a single endpoint assuming that we would have access to a backend component. I *may* need to revert to the previous approach of supporting different endpoints, in which case I would use this dead code.
